packages/core/nuxt-theme-module/theme/pages/Category.vue

```diff
@@ -250,111 +249,51 @@ import {
-import { getFiltersFromUrl, getFiltersForUrl } from '~/helpers/filters';
+import { computed, ref, onMounted } from '@vue/composition-api';
+import { useFacet, useCart, useWishlist, facetGetters, productGetters } from '<%= options.generate.replace.composables %>';
+import createThemeHelpers from '~/helpers/ui';
 import { onSSR } from '@vue-storefront/core';
 import Filters from '../components/Filters';
 
-const perPageOptions = [20, 40, 100];
-
-// TODO: to be implemented in https://github.com/DivanteLtd/next/issues/211
-const fallbackBreadcrumbs = [
-  { text: 'Home', route: { link: '#' } },
-  { text: 'Women', route: { link: '#' } }
-];
-
 export default {
   transition: 'fade',
   setup(props, context) {
-    const { query } = context.root.$route;
+    const th = createThemeHelpers();
     onMounted(() => context.root.$scrollTo(context.root.$el, 2000));
+    const isGridView = ref(true);
+    const isFilterSidebarOpen = ref(false);
 
-    const { categories, search, loading: categoriesLoading } = useCategory('categories');
-    const {
-      products: categoryProducts,
-      totalProducts,
-      search: productsSearch,
-      loading: productsLoading,
-      availableFilters,
-      availableSortingOptions
-    } = useProduct('categoryProducts');
     const { loadCart, addToCart, isOnCart } = useCart();
     const { addToWishlist } = useWishlist();
+    const { searchData, search, loading } = useFacet();
 
-    const currentPage = ref(parseInt(query.page, 10) || 1);
-    const itemsPerPage = ref(parseInt(query.items, 10) || perPageOptions[0]);
-    const sortBy = ref(query.sort || (availableSortingOptions?.value && availableSortingOptions?.value[0] ? availableSortingOptions.value[0]?.value : null));
-    const filters = ref(null);
-    const loading = computed(() => categoriesLoading.value || productsLoading.value);
-    const productsSearchParams = computed(() => ({
-      catId: (categories.value[0] || {}).id,
-      page: currentPage.value,
-      perPage: itemsPerPage.value,
-      filters: filters.value,
-      sort: sortBy.value
-    }));
+    const products = computed(() => facetGetters.getResults(searchData.value));
+    const categoryTree = computed(() => facetGetters.getCategoryTree(searchData.value));
+    const breadcrumbs = computed(() => facetGetters.getBreadcrumbs(searchData.value));
+    const sortBy = computed(() => facetGetters.getSortOptions(searchData.value));
+    const facets = computed(() => facetGetters.getGroupedFacets(searchData.value, ['color', 'size']));
+    const pagination = computed(() => facetGetters.getPaginationInfo(searchData.value));
 
     onSSR(async () => {
-      await search(getCategorySearchParameters(context));
-      await productsSearch(productsSearchParams.value);
-      filters.value = getFiltersFromUrl(context, availableFilters.value);
-      await productsSearch(productsSearchParams.value);
+      await search(th.getFacets());
```

Comment:
(theme helpers are completely different discussion but just leaving it here so i wont forget)

`getFacets()` is not descriptive at all, is it `getFacetsFromURL` or what? why this method has the same name as facet getter and why its not aprametrized (pure) ;d